w3/streaming-multihash/src/hasher.js

```diff
@@ -0,0 +1,56 @@
+import * as API from './type.js'
+import * as Digest from 'multiformats/hashes/digest'
+
+/**
+ * @template {string} Name
+ * @template {number} Code
+ * @param {object} options
+ * @param {Name} options.name
+ * @param {Code} options.code
+ * @param {{create: () => API.RawHasher}} options.hasher
+ */
+
+export const derive = ({ name, code, hasher }) => {
+  /**
+   * @returns {API.StreamingHasher<Code>}
+   */
+  const create = () => new StreamingHasher(code, hasher.create())
+
+  /**
+   * @param {API.AwaitIterable<Uint8Array>} stream
+   */
+  const digestStream = async (stream) => {
+    const hash = hasher.create()
+    for await (const chunk of stream) {
+      hash.update(chunk)
+    }
+    return Digest.create(code, hash.digest())
+  }
+
+  return { name, code, create, digestStream }
```

Comment:
```suggestion
  return { name, code, hasher: new StreamingHasher(code, hasher) }
```

this is all it is, `derives` wraps any hasher in a multihash based hasher with support for `AwaitIterable` directly

Reply:
actually if we add the `name` as a property to `StreamingHasher` we can just return `new StreamingHasher`